arrows/vxl/pixel_feature_extractor.cxx

```diff
@@ -118,19 +118,32 @@ ::concatenate_images( std::vector< vil_image_view< pix_t > > filtered_images )
     return {};
   }
 
-  auto const ni = filtered_images.at( 0 ).ni();
-  auto const nj = filtered_images.at( 0 ).nj();
+  auto const ni = filtered_images.at( 0 )->ni();
+  auto const nj = filtered_images.at( 0 )->nj();
   vil_image_view< pix_t > concatenated_planes{ ni, nj, total_planes };
 
   // Concatenate the filtered images into a single output
   unsigned current_plane{ 0 };
 
-  for( auto const& image : filtered_images )
+  for( auto const& filtered_image_ptr : filtered_images )
   {
-    for( unsigned i{ 0 }; i < image.nplanes(); ++i )
+    vil_image_view< pix_t > filtered_image;
+    if( filtered_image_ptr->pixel_format() ==
+        concatenated_planes.pixel_format() )
+    {
+      filtered_image = filtered_image_ptr;
+    }
+    else
+    {
+      // Cast the image pointer into the concrete type we want
+      filtered_image = vil_convert_cast( pix_t(), filtered_image_ptr );
+    }
+
+    for( unsigned i{ 0 }; i < filtered_image.nplanes(); ++i )
     {
-      vil_plane( concatenated_planes,
-                 current_plane ).deep_copy( vil_plane( image, i ) );
+      auto output_plane = vil_plane( concatenated_planes, current_plane );
+      vil_image_view< pix_t > input_plane = vil_plane( filtered_image, i );
```

Comment:
Using `auto` in the preceding line but not here seems a little incongruous. Is there a necessary typecast happening here? If yes, it might be nice to make that more obvious by writing instead `auto input_plane = vil_image_view< pix_t >{ vil_plane( filtered_image, i ) };`.

Reply:
Good catch, there was no need for the explicit type.